core/templates/pages/exploration-player-page/services/extract-image-filenames-from-model.service.ts

```diff
@@ -191,26 +195,31 @@ export class ExtractImageFilenamesFromModelService {
     let collapsibleTagList = dummyDocument.getElementsByTagName(
       'oppia-noninteractive-collapsible');
     for (let i = 0; i < collapsibleTagList.length; i++) {
-      let contentWithValue = JSON.parse(
-        collapsibleTagList[i].getAttribute('content-with-value'));
-      let collapsibleDocument = (
-        new DOMParser().parseFromString(contentWithValue, 'text/html'));
-      imageTagLists.push(
-        collapsibleDocument.getElementsByTagName('oppia-noninteractive-image'));
+      let attribute = collapsibleTagList[i].getAttribute('content-with-value');
+      if (attribute !== null) {
```

Comment:
Why is this done? When can attribute be `null`?

Reply:
Created an attribute variable that gets the given attribute from a collapsible. If the given attribute does not exist, then attribute is null which means skip the current collapsible. If attribute is not null, then add images that are embedded in the given collapsible.